### PR TITLE
docs(ReadMe): update readme file for sourceloop core

### PR DESCRIPTION
**Pull request by Surbhi-sharma1**
gh-254

## Description
update readme file for sourceloop core

Fixes # (issue)
#254 
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/core/README.md [hunk 1]
```diff
@@ -27,8 +27,300 @@
...
+    this.application.component(BearerVerifierComponent);
+}
+```
+## Logger-Extension
+
+### Usage
+
+- Create a new Loopback4 Application (If you don't have one already)
+  `lb4 testapp`
+- Install the authentication service
```
Comment (yeshamavani) [inline packages/core/README.md:121]: Authentication?

File: packages/core/README.md [hunk 2]
```diff
@@ -27,6 +27,308 @@
 
 that are used throughout the service catalog.
 
+### Adapters
+
+Adapters works as a bridge between two incompatible interfaces. This type of design pattern comes under structural pattern as this pattern combines the capability of two independent interfaces.
```
Comment (yeshamavani) [inline packages/core/README.md:32]: no need to explain adapter pattern, if you feel the need can give links to the article that explains it

File: packages/core/README.md [hunk 3]
```diff
@@ -27,6 +27,308 @@
...
+
+### Usage
+
+- Create a new Loopback4 Application (If you don't have one already)
+  `lb4 testapp`
+- Install the authentication service
+  `npm i @sourceloop/core`
+- Configure `@sourceloop/core` component to include `LoggerExtensionComponent` -
+  ```typescript
+  this.providers = {[LOGGER.BINDINGS.LOG_ACTION.key]: LoggerProvider};
```
Comment (yeshamavani) [inline packages/core/README.md:125]: We bind the provider like this in application 
this.bind(LOGGER.BINDINGS.LOG_ACTION.toProvider(
      LoggerProvider,
    );

File: packages/core/README.md [hunk 4]
```diff
@@ -27,6 +27,308 @@
...
+    name: 'modified_by',
+  })
+  modifiedBy?: string;
+}
+```
+
+In order to use this in your application:
+
+- Extend models with UserModifiableEntity Class replacing entity adding two attributes to the
+  model Class i.e. createdBy and modifiedBy. The column names needed to be there in DB within that table are 'createdBy' and 'modifiedBy'.
```
Comment (yeshamavani) [inline packages/core/README.md:220]: ?

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: packages/core/README.md (hunk 2 above)
Comment (Surbhi-sharma1) [reply packages/core/README.md:32]: ok

File: packages/core/README.md (hunk 1 above)
Comment (Surbhi-sharma1) [reply packages/core/README.md]: changes done

File: packages/core/README.md (hunk 3 above)
Comment (Surbhi-sharma1) [reply packages/core/README.md]: done

File: packages/core/README.md (hunk 4 above)
Comment (Surbhi-sharma1) [reply packages/core/README.md]: changes done

File: packages/core/README.md [hunk 5]
```diff
@@ -27,6 +27,304 @@
...
+
+## Swagger Authentication Component
+
+A Loopback Component that adds an authenticating middleware for Rest Explorer
+
+### Installation
+
+```bash
+
+npm i @sourceloop/authentication-service
```
Comment (akshatdubeysf) [inline packages/core/README.md:135]: This seems wrong?

File: packages/core/README.md [hunk 6]
```diff
@@ -27,6 +27,304 @@
...
+
+For all the enums provided in here, see `packages/core/src/enums`.
+
+### Mixins
+
+You can find documentation for mixins. [here](/src/mixins)
+
+### Models
+
+A model describes business domain objects(shape of data) to be persisted in the database, for example, Customer, Address, and Order. It usually defines a list of properties with name, type, and other constraints.We use decorators @model and @property to annotate or modify the Class members and Class respectively which helps in manipulating the metadata or even integrate with JSON Schema generation.
```
Comment (akshatdubeysf) [inline packages/core/README.md:196]: You also need to describe the different models provided like BaseEntity, etc and the fields provided by them

File: packages/core/README.md [hunk 7]
```diff
@@ -27,6 +27,304 @@
...
+You can find documentation for the providers available .[here](/src/providers)
+
+### Repositories
+
+A Repository represents a specialized Service Interface that provides strong-typed data access (for example, CRUD) operations of a domain model against the underlying database or service
+Repositories are adding behavior to Models. Models describe the shape of data, Repositories provide behavior like CRUD operations.
+
+![Repository](https://loopback.io/pages/en/lb4/imgs/repository.png)
+
+DefaultUserModifyCrudRepository -A repository Class which will provide CRUD operations for default user modifications.
```
Comment (akshatdubeysf) [inline packages/core/README.md:231]: DefaultUserModifyCrudRepository also add modified by fields, you need to mention that as well

File: packages/core/README.md [hunk 8]
```diff
@@ -27,6 +27,304 @@
...
+
+![Connector](https://loopback.io/images/9830486.png)
+
+### Sequence Handlers
+
+Sourceloop Core has sequences that can be used in application providing casbin-secure sequence,secure-sequence and service sequence.
+
+## Usage
+
+-import the sequence you want to use.for example,import ServiceSequence in your application component file.
```
Comment (akshatdubeysf) [inline packages/core/README.md:279]: Need to explain what is service sequence, what is secure sequence, etc

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

File: packages/core/README.md (hunk 7 above)
Comment (Surbhi-sharma1) [reply packages/core/README.md:231]: ok sir

File: packages/core/README.md (hunk 6 above)
Comment (Surbhi-sharma1) [reply packages/core/README.md:196]: done

File: packages/core/README.md (hunk 5 above)
Comment (Surbhi-sharma1) [reply packages/core/README.md]: done

File: packages/core/README.md (hunk 7 above)
Comment (Surbhi-sharma1) [reply packages/core/README.md]: done

**Review by akshatdubeysf — APPROVED** [review]

**Pull request merged**